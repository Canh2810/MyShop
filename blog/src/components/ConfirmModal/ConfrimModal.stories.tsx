import { StoryObj, Meta } from '@storybook/react'

import ConfirmModal from '.'

export default {
  title: 'Components/ConfirmModal',
  component: ConfirmModal,
} as Meta

type Story = StoryObj<typeof ConfirmModal>

export const Default: Story = {
  args: {
    title: 'Confirm this modal',
    onSubmit: () => alert('Submit modal'),
    onClose: () => alert('Close modal'),
  },
}
