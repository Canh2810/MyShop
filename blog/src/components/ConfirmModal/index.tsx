import { memo } from 'react'
import { Button, Typography } from '..'
import { ButtonVariants } from '@/types'

interface ConfirmModalProps {
  title: string
  isLoading?: boolean
  onSubmit: () => void
  onClose: () => void
}

const ConfirmModal = ({
  title,
  isLoading,
  onSubmit,
  onClose,
}: ConfirmModalProps) => {
  return (
    <div className="fixed top-0 left-0 w-full h-full bg-light-700 z-[100] ">
      <div className=" top-0 left-0 w-full h-full" onClick={onClose}></div>
      <div className="bg-white dark:bg-dark-300 rounded-xl p-6 absolute top-2/4 left-2/4 translate-y-[-50%] translate-x-[-50%] z-[101] text-center">
        <Typography>{title}</Typography>
        <div className="flex items-center gap-7 justify-center mt-10">
          <Button onClick={onClose}>Cancel</Button>
          <Button
            variant={ButtonVariants.Container}
            isLoading={isLoading}
            onClick={onSubmit}
          >
            Confirm
          </Button>
        </div>
      </div>
    </div>
  )
}

export default memo(ConfirmModal)
