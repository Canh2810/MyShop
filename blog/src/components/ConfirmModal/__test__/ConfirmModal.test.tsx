import React from 'react'
import { render, waitFor, screen } from '@testing-library/react'
import ConfirmModal from '..'
import userEvent from '@testing-library/user-event'

const onSubmitMock = jest.fn()
const onCloseMock = jest.fn()

const setup = () =>
  render(
    <ConfirmModal
      title="This is modal"
      onSubmit={onSubmitMock}
      onClose={onCloseMock}
    />,
  )

describe('ConfirmModal', () => {
  it('renders the modal with the correct', () => {
    const { container } = setup()
    expect(container).toMatchSnapshot()
  })

  it('calls the onClose callback when clicking the Cancel button', async () => {
    setup()
    await userEvent.click(screen.getByText('Cancel'))
    expect(onCloseMock).toHaveBeenCalled()
  })

  it('calls the onSubmit callback when clicking the Confirm button', async () => {
    setup()
    await userEvent.click(screen.getByText('Confirm'))
    await waitFor(() => expect(onSubmitMock).toHaveBeenCalled())
  })
})
