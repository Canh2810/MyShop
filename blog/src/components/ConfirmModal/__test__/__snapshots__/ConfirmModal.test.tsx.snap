// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal renders the modal with the correct 1`] = `
<div>
  <div
    class="fixed top-0 left-0 w-full h-full bg-light-700 z-[100] "
  >
    <div
      class=" top-0 left-0 w-full h-full"
    />
    <div
      class="bg-white dark:bg-dark-300 rounded-xl p-6 absolute top-2/4 left-2/4 translate-y-[-50%] translate-x-[-50%] z-[101] text-center"
    >
      <p
        class="font-tertiary text-sm font-normal leading-4 text-dark-100 dark:text-light-400 "
      >
        This is modal
      </p>
      <div
        class="flex items-center gap-7 justify-center mt-10"
      >
        <button
          class=" rounded-md hover:opacity-70 disabled:opacity-40 px-5 py-3 bg-white dark:bg-dark-300 text-dark-50 border border-solid border-light-700 "
        >
          Cancel
        </button>
        <button
          class=" rounded-md hover:opacity-70 disabled:opacity-40 px-5 py-3 bg-primary-500 text-white "
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
